autoload/which_key.vim

```diff
@@ -216,12 +218,21 @@ function! s:echo_prompt() abort
 endfunction
 
 function! s:has_children(input) abort
+  let cnt = 0
   if index(s:REQUIRES_REGEX_ESCAPE, a:input) != -1
-    let group = map(keys(s:runtime), {_,v -> v =~# '^\'.a:input})
+    let regex = '^\'.a:input
   else
-    let group = map(keys(s:runtime), {_,v -> v =~# '^'.a:input})
+    let regex = '^'.a:input
   endif
-  return len(filter(group, 'v:val == 1')) > 1
+  for each in keys(s:runtime)
+    if each =~# regex
+      let cnt += 1
+      if cnt > 1
+        return 1
+      endif
+    endif
+  endfor
+  return 0
 endfunction
```

Comment:
Unsure in which path lambda was initially added, but `+lambda` is available since 7.4.2121, therefore it's definitely earlier than 7.4.2121. I'm not against this change, but can you keep the existing version when lambda is usable?

```vim
if has('+lambda')
    function! s:has_children(input) abort
    " existing version
    endfunction
else
    function! s:has_children(input) abort
    " legacy
    endfunction
endif
    
```